client/websocket_client.go

```diff
@@ -0,0 +1,193 @@
+	// check no errors are reported on the websocket 
+	if err := erisNodeWebsocketClient.assertNoErrors(); err != nil {
+		return nil, err
+	}
+
+	// Setup the confirmation channel to be returned
+	confirmationChannel := make(chan Confirmation, 1)
+	var latestBlockHash []byte
+
+	eid := txs.EventStringAccInput(inputAddr)
+
+	// Read the incoming events
+	go func() {
+		var err error
+		for {
+			resultBytes := <- erisNodeWebsocketClient.tendermintWebsocket.ResultsCh
+			result := new(ctypes.ErisDBResult)
+			if wire.ReadJSONPtr(result, resultBytes, &err); err != nil {
+				// keep calm and carry on
+				log.Errorf("eris-client - Failed to unmarshal json bytes for websocket event: %s", err)
+				continue
+			}
+			
+			event, ok := (*result).(*ctypes.ResultEvent)
+			if !ok {
+				// keep calm and carry on
+				log.Error("eris-client - Failed to cast to ResultEvent for websocket event")
+				continue
+			}
+			
+			blockData, ok := event.Data.(txs.EventDataNewBlock)
+			if ok {
+				latestBlockHash = blockData.Block.Hash()
+				log.WithFields(log.Fields{
+					"new block": blockData.Block,
+					"latest hash": latestBlockHash,
+				}).Debug("Registered new block")
+				continue
+			}
+			
+			// we don't accept events unless they came after a new block (ie. in)
+			if latestBlockHash == nil {
+				continue
+			}
+
+			if event.Event != eid {
+				log.Warnf("Received unsolicited event! Got %s, expected %s\n", event.Event, eid)
+				continue
+			}
+
+			data, ok := event.Data.(txs.EventDataTx)
+			if !ok {
+				// We are on the lookout for EventDataTx
+				confirmationChannel <- Confirmation{
+					BlockHash: latestBlockHash,
+					Event: nil,
+					Exception: fmt.Errorf("response error: expected result.Data to be *types.EventDataTx"),
+					Error: nil,
+				}
+				return
+			}
+
+			if !bytes.Equal(txs.TxHash(chainId, data.Tx), txs.TxHash(chainId, tx)) {
+				log.WithFields(log.Fields{
+					// TODO: consider re-implementing TxID again, or other more clear debug
+					"received transaction event": txs.TxHash(chainId, data.Tx),
+				}).Debug("Received different event")
+				continue
+			}
+
+			if data.Exception != "" {
+				confirmationChannel <- Confirmation{
+					BlockHash: latestBlockHash,
+					Event: &data,
+					Exception: fmt.Errorf("Transaction confirmed with exception:", data.Exception),
+					Error: nil,
+				}
+				return
+			}
+			
+			// success, return the full event and blockhash and exit go-routine
+			confirmationChannel <- Confirmation{
+				BlockHash: latestBlockHash,
+				Event: &data,
+				Exception: nil,
+				Error: nil,
+			}
+			return
+		}
+
+	}()
+
+	// TODO: [ben] this is a draft implementation as resources on time.After can not be
+	// recovered before the timeout.  Close-down timeout at success properly.
+	timeout := time.After(time.Duration(MaxCommitWaitTimeSeconds) * time.Second)
```

Comment:
Ha! To my victory.  The mint-client code had time.Tick which I thought was more concerning: if used as a command line, then that is fine as program terminates; but we use it as a library in e-pm and that is a problematic use (of time.Tick here);  So I changed it to time.After, so indeed that should saturate the resources used on a 10 second window; but will still pile up these go-routines if many calls are made per second.

For now this can pass as sufficient, but for load testing, I think this might be a point of concern